src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -144,7 +144,7 @@ const useCoreChatSystemHook = () => {
     const channelType: {[key: string]: ChannelType} = {
       messaging: isAnonymous ? 'ANON_PM' : 'PM',
       group: 'GROUP',
-      topics: 'TOPIC'
+      topics: isAnonymous ? 'ANON_TOPIC' : 'TOPIC'
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is logical and consistent with the existing code. It adds an anonymous topic type to the `channelType` mapping, which will be used if `isAnonymous` is true. This seems like a good way to handle different types of channels based on whether the user is anonymous or not.
```diff
- topics: 'TOPIC'
+ topics: isAnonymous ? 'ANON_TOPIC' : 'TOPIC'
```
This change should improve the flexibility of the chat system by allowing for anonymous topics. However, it's important to ensure that the rest of the codebase can handle this new 'ANON_TOPIC' channel type correctly.


<!-- This is an auto-generated comment by OSS CodeRabbit -->